Themis/ViewModels/Course/CourseViewModel.swift

```diff
@@ -54,6 +54,9 @@ class CourseViewModel: ObservableObject {
             if shownCourseID == nil {
                 shownCourseID = self.courses.first?.id
             }
+            for i in 0...courses.count - 1 {
+                try await courses[i].fetchProgrammingExercises(courseId: courses[i].id)
+            }
```

Comment:
this will fail if courses.count is 0 since then loverBound > upperbound
replace with 0..<courses.count